yt/visualization/plot_window.py

```diff
@@ -1175,7 +1207,10 @@ def _setup_plots(self):
                         )
                     else:
                         ymin, ymax = (float(y) for y in extenty)
-                    self.plots[f].image.set_extent((xmin, xmax, ymin, ymax))
+                    new_extent = [xmin, xmax, ymin, ymax]
+                    if swap_axes:
+                        new_extent = [ymin, ymax, xmin, xmax]
```

Comment:
no reason not to preserve the existing type
```suggestion
                    new_extent = (xmin, xmax, ymin, ymax)
                    if swap_axes:
                        new_extent = (ymin, ymax, xmin, xmax)
```